sustainability/src/main/java/edu/asu/diging/sustainability/web/search/ResearcherSearchController.java

```
@@ -2,33 +2,56 @@
 import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
 import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
 
 import edu.asu.diging.sustainability.core.data.ConceptRepository;
+import edu.asu.diging.sustainability.core.exception.NotAValidResourceException;
+import edu.asu.diging.sustainability.core.model.IAnnotation;
 import edu.asu.diging.sustainability.core.model.IConcept;
+import edu.asu.diging.sustainability.core.model.IResource;
 import edu.asu.diging.sustainability.core.model.impl.Concept;
 import edu.asu.diging.sustainability.core.service.IAnnotationManager;
+import edu.asu.diging.sustainability.core.service.IResourceManager;
 
 @Controller
 public class ResearcherSearchController {
-    
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
     @Autowired
     private IAnnotationManager annotationManager;
-    
+
     @Autowired
     private ConceptRepository conceptRepo;
 
-    @RequestMapping(value="/perspective/search", method=RequestMethod.POST)
+    @Autowired
+    private IResourceManager resourceManager;
+
+    @RequestMapping(value = "/perspective/search", method = RequestMethod.POST)
     public String search(Model model, @RequestParam("selectedConcepts") String[] selectedConcepts) {
-        model.addAttribute("results", annotationManager.findTextsForConcepts(selectedConcepts));
-        
+        Map<String, List<IAnnotation>> URIList =
+                annotationManager.findTextsForConcepts(selectedConcepts);
+        List<IResource> resourceList = new ArrayList<IResource>();
+        for (String uri : URIList.keySet()) {
+            try {
+                resourceList.add(resourceManager.getResource(uri));
+            } catch (NotAValidResourceException e) {
+                logger.error("Resource " + uri + " is not a valid one.");
```

don't swallow the exception!

---

not addressed